convert_hf_to_gguf.py

```diff
@@ -2702,6 +2703,80 @@ class StarCoder2Model(Model):
     model_arch = gguf.MODEL_ARCH.STARCODER2
 
 
+@Model.register("Rwkv6ForCausalLM")
+class RwkvModel(Model):
+    model_arch = gguf.MODEL_ARCH.RWKV
+
+    def set_vocab(self):
+        assert (self.dir_model / "rwkv_vocab_v20230424.txt").is_file()
+        vocab_size = self.hparams.get("vocab_size", 65536)
+
+        tokens: list[bytes] = ['<s>'.encode("utf-8")]
+        toktypes: list[int] = [gguf.TokenType.CONTROL]
+
+        with open(self.dir_model / "rwkv_vocab_v20230424.txt", "r", encoding="utf-8") as f:
+            lines = f.readlines()
+            for line in lines:
+                parts = line.split(' ')
+                assert len(parts) >= 3
+                _, token, token_len = int(parts[0]), ast.literal_eval(' '.join(parts[1:-1])), int(parts[-1])
+                token = token.encode("utf-8") if isinstance(token, str) else token
+                assert isinstance(token, bytes)
+                assert len(token) == token_len
+                token_text: str = ""
+                for b in token:
+                    token_text += f"\\x{b:02x}"
+                tokens.append(token_text.encode("utf-8"))
```

Comment:
Why are the token bytes converted to an escape sequence rather than saved directly?

Reply:
That's because the rwkv_world tokenizer contains incomplete UTF-8 bytes in decoded sequences. However, llama.cpp seems to require valid UTF-8 texts.

Reply:
Noted. The manual escaping could be avoided by doing `token_text = str(token)[2:-1]`.

```
>>> b = bytes("Testこんにちは\n123", "utf-8")
>>> b
b'Test\xe3\x81\x93\xe3\x82\x93\xe3\x81\xab\xe3\x81\xa1\xe3\x81\xaf\n123'
>>> str(b)
"b'Test\\xe3\\x81\\x93\\xe3\\x82\\x93\\xe3\\x81\\xab\\xe3\\x81\\xa1\\xe3\\x81\\xaf\\n123'"
>>> str(b)[2:-1]
'Test\\xe3\\x81\\x93\\xe3\\x82\\x93\\xe3\\x81\\xab\\xe3\\x81\\xa1\\xe3\\x81\\xaf\\n123'
>>> print(str(b)[2:-1])
Test\xe3\x81\x93\xe3\x82\x93\xe3\x81\xab\xe3\x81\xa1\xe3\x81\xaf\n123
```

As far as I know, `str(bytes_object)` will return a string with only normal ASCII characters, `\t`, `\n`, `\r`, or hex escape sequences.

Reply:
Tested and it looks good to me. Thanks for your help!